### PR TITLE
feat: add hero section

### PR DESCRIPTION
**Pull request by ishita-1305**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [ ] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.
- [ ] I have added necessary documentation (if appropriate)

### Description
Please describe your pull request.

❤️Thank you!

### Post merge checklist
- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

**Comment by Lucif3r-in** [comment]
Make the requested changes @ishita-1305

**Comment by Lucif3r-in** [comment]
@ishita-1305 `5.0` should be right aligned and the text should be justified. The `Hack` also gets smaller than `NITR` on smaller screens

**Comment by SirSimon162** [comment]
@ishita-1305 
- Fix the typography in the hero section for mobile view.
- Fix the padding around 'Hack' image in mobile view.
- Fix the mobile view as per the design.
- Focus on alignment.

**Review by ishita-1305 — COMMENTED** [review]
-fixed alignment in mobile screen
-fixed the padding around 'Hack' image in mobile view.
-fixed the mobile view as per the design.
-removed the buttons

**Comment by Lucif3r-in** [comment]
Keep the width of divs containing HackNITR, 5.0 and the text beneath same

**Comment by Lucif3r-in** [comment]
Add `cursor:pointer` to buttons and make the `flex-directiion:column` for smaller screens

**Comment by Lucif3r-in** [comment]
@ishita-1305 Make changes according to figma file.

**Comment by Lucif3r-in** [comment]
Resolve merge conflicts @ishita-1305

File: src/components/HeroSection/styles.jsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    justify-center
+    items-center
+    text-white
+    flex-col
+    overflow-x-hidden
+    overflow-y-hidden
+    w-full
+    `}
+`;
+export const heroBelow = styled.div`
```
Comment (anmol5936) [inline src/components/HeroSection/styles.jsx:31]: remove heroBelow from here.

File: src/components/HeroSection/HeroSection.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+export default function Hero() {
+  return (
+    <SectionContainer >
+      <HeroBody >
+        <HeroTitle className='hero h-100'>
+          <HeroInside>
+            <HACKimg src={hero.imgs.hack.src} alt={hero.imgs.hack.alt}/>
+            <NITRimg src={hero.imgs.nitr.src} alt={hero.imgs.nitr.alt}/>
+          </HeroInside>
+          <heroBelow className='mx-8 md:max-w-xl lg:max-w-3xl'>
```
Comment (anmol5936) [inline src/components/HeroSection/HeroSection.jsx:29]: no need to use heroBelow tag instead use another name

File: src/components/HeroSection/HeroSection.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          <HeroInside>
+            <HACKimg src={hero.imgs.hack.src} alt={hero.imgs.hack.alt}/>
+            <NITRimg src={hero.imgs.nitr.src} alt={hero.imgs.nitr.alt}/>
+          </HeroInside>
+          <contentBlock className='mx-8 md:max-w-xl lg:max-w-3xl'>
+            <HeroHeading className='flex border-white w-full'>
+              <HeroHeadingBase>
+                <FIVEimg src={hero.imgs.five.src}/>
+              </HeroHeadingBase>
+            </HeroHeading>
```
Comment (Lucif3r-in) [inline src/components/HeroSection/HeroSection.jsx:33]: Why do we need two wrappers wrapping the same component?

File: src/components/HeroSection/styles.jsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    justify-center
+    gap-2
+    items-center
+    flex-col
+    px-36
+    md:px-12
+    // lg:px-36
+    mx-10
+    mt-32
+
```
Comment (Lucif3r-in) [inline src/components/HeroSection/styles.jsx:44]: run prettier once in the file

File: src/components/shared/DevfolioButton.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable max-len */
```
Comment (Lucif3r-in) [inline src/components/shared/DevfolioButton.jsx:1]: Remove unwanted comments

**Review by Lucif3r-in — CHANGES_REQUESTED** [review]

File: src/components/HeroSection/styles.jsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    lg:w-96
+ `}
+`;
+
+export const HeroHeadingBase = styled.div`
+  ${tw`
+      flex
+      justify-end
+      w-full
+
```
Comment (Lucif3r-in) [inline src/components/HeroSection/styles.jsx:91]: Run prettier once in all these files please

**Review by Lucif3r-in — CHANGES_REQUESTED** [review]

**Comment by Lucif3r-in** [comment]
LGTM🚀

File: src/pages/index.js [hunk 7]
```diff
@@ -46,10 +46,11 @@ const HomePage = ({ location }) => {
         <script type='application/ld+json'>{JSON.stringify(breadcrumbsData)}</script>
       </Helmet>
       <SEO />
-
+
```
Comment (Lucif3r-in) [inline src/pages/index.js:49]: Remove this change

File: src/pages/index.js [hunk 8]
```diff
@@ -46,10 +46,11 @@ const HomePage = ({ location }) => {
         <script type='application/ld+json'>{JSON.stringify(breadcrumbsData)}</script>
       </Helmet>
       <SEO />
-
+        
       <Container>
         <Heading1>Hello World</Heading1>
       </Container>
+
```
Comment (Lucif3r-in) [inline src/pages/index.js:53]: Extra line

File: src/styles/global.css [hunk 9]
```diff
@@ -9,6 +9,7 @@
 *:after {
   box-sizing: border-box;
   -webkit-font-smoothing: antialiased;
+
```
Comment (Lucif3r-in) [inline src/styles/global.css:12]: Extra line

File: src/components/shared/DiscordButton.jsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+function DiscordButton() {
+  return (
+
+    <div className="btn" >
+  <div className="register_btn">
+    <button type="submit">D</button>
+  </div>
+</div>
+
```
Comment (Lucif3r-in) [inline src/components/shared/DiscordButton.jsx:12]: Remove extra spaces run prettier

File: src/components/shared/DevfolioButton.jsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+import "./DevfolioBtn.css"
+
+const RegisterButton = () =>{
+  {
+    const [clicked, setClicked] = useState(false);
+
+    const handleButtonClick = () => {
+      setClicked(true);
+    };
+
```
Comment (Lucif3r-in) [inline src/components/shared/DevfolioButton.jsx:11]: Run prettier in all the files once

File: src/components/shared/DevfolioBtn.css [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+@media (max-width: 768px) {
+  button {
+    font-size: 15px;
+    padding: 15px 40px;
+    box-shadow: #C48DFF 1px 1px, #FF8CD9 2px 2px, #35E1FF 3px 3px;
+    border-radius: 10px;
+  }
+}
+
+
```
Comment (Lucif3r-in) [inline src/components/shared/DevfolioBtn.css:65]: Remove extra space

File: src/components/shared/DevfolioBtn.css [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+  border-radius: 15px;
+  transition: box-shadow 0.3s;
+}
+
+button :active {
+  top: 10px;
+  background-color: #ffffff;
+  box-shadow: none;
+  transform:translateY(3px 3px)
+
```
Comment (Lucif3r-in) [inline src/components/shared/DevfolioBtn.css:35]: Here too

**Review by Lucif3r-in — CHANGES_REQUESTED** [review]

**Comment by Lucif3r-in** [comment]
LGTM🚀🚀

**Pull request merged**